packages/repository/src/connector.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {AnyType} from './common';
+import {Model} from './model';
+
+/**
+ * Common properties/operations for connectors
+ */
+export interface Connector {
+  name: string; // Name/type of the connector
+  configModel?: Model; // The configuration model
+  interfaces?: string[]; // A list of interfaces implemented by the connector
+  connect(): Promise<AnyType>; // Connect to the underlying system
+  disconnect(): Promise<AnyType>; // Disconnect from the underlying system
+  ping(): Promise<AnyType>; // Ping the underlying system
```

Comment:
`Promise<AnyType>` means that the method returns a promise that can be resolved with any value. I don't see how is that going to be useful for callers of these three methods (`connect`, `disconnect` and `ping`) when each connector can return whatever it wants.

I am proposing to change return type of these three functions to `Promise<void>`.

If you disagree then please explain your reasoning so that we can discuss.